astropy/convolution/core.py

```diff
@@ -88,42 +85,66 @@ def center(self):
         """
         return [axes_size // 2 for axes_size in self._array.shape]
 
+    def calculate_normalization(self, mode='integral'):
+        """
+        Calculate the kernel normalization factor.
+
+        Parameters
+        ----------
+        mode : {'integral', 'peak'}
+            One of the following modes:
+                * 'integral' (default)
+                    Kernel is normalized such that its integral = 1.
+                * 'peak'
+                    Kernel is normalized such that its peak = 1.
+        """
+
+        if mode == 'integral':
+            if self._array.sum() == 0:
+                self._normalization = np.inf
+            else:
+                self._normalization = 1. / self._array.sum()
+        elif mode == 'peak':
+            self._normalization = 1. / self._array.max()
```

Comment:
I think `self._normalization` should be set to `1. / self._array.sum()` regardless whether the mode is `peak` or `integral` (see comment below).